controllers/tests/suite_test.go

```diff
@@ -82,116 +83,122 @@ func TestAPIs(t *testing.T) {
+		Expect(err).NotTo(HaveOccurred())
 
-	err = controllers.SetupKafkaUserWithManager(mgr, true)
-	Expect(err).NotTo(HaveOccurred())
+		err = controllers.SetupKafkaUserWithManager(mgr, true)
+		Expect(err).NotTo(HaveOccurred())
 
-	kafkaClusterCCReconciler := controllers.CruiseControlTaskReconciler{
-		Client: mgr.GetClient(),
-		Scheme: mgr.GetScheme(),
-		Log:    ctrl.Log.WithName("controller").WithName("CruiseControlTask"),
-	}
+		kafkaClusterCCReconciler := controllers.CruiseControlTaskReconciler{
+			Client: mgr.GetClient(),
+			Scheme: mgr.GetScheme(),
+			Log:    ctrl.Log.WithName("controller").WithName("CruiseControlTask"),
+		}
 
-	err = controllers.SetupCruiseControlWithManager(mgr).Complete(&kafkaClusterCCReconciler)
-	Expect(err).NotTo(HaveOccurred())
+		err = controllers.SetupCruiseControlWithManager(mgr).Complete(&kafkaClusterCCReconciler)
+		Expect(err).NotTo(HaveOccurred())
 
-	// +kubebuilder:scaffold:builder
+		// +kubebuilder:scaffold:builder
 
-	go func() {
-		ctrl.Log.Info("starting manager")
-		err = mgr.Start(ctrl.SetupSignalHandler())
-		Expect(err).ToNot(HaveOccurred())
-	}()
+		go func() {
+			ctrl.Log.Info("starting manager")
+			err = mgr.Start(ctrl.SetupSignalHandler())
+			Expect(err).ToNot(HaveOccurred())
+		}()
 
-	k8sClient, err = client.New(cfg, client.Options{Scheme: mgr.GetScheme()})
-	Expect(err).ToNot(HaveOccurred())
-	Expect(k8sClient).ToNot(BeNil())
+		k8sClient, err = client.New(cfg, client.Options{Scheme: mgr.GetScheme()})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(k8sClient).ToNot(BeNil())
 
-	crd := &apiextensionsv1beta1.CustomResourceDefinition{}
+		crd := &apiextensionsv1beta1.CustomResourceDefinition{}
 
-	err = k8sClient.Get(context.TODO(), types.NamespacedName{Name: "kafkaclusters.kafka.banzaicloud.io"}, crd)
-	Expect(err).NotTo(HaveOccurred())
-	Expect(crd.Spec.Names.Kind).To(Equal("KafkaCluster"))
+		err = k8sClient.Get(context.TODO(), types.NamespacedName{Name: "kafkaclusters.kafka.banzaicloud.io"}, crd)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(crd.Spec.Names.Kind).To(Equal("KafkaCluster"))
 
-	err = k8sClient.Get(context.TODO(), types.NamespacedName{Name: "kafkatopics.kafka.banzaicloud.io"}, crd)
-	Expect(err).NotTo(HaveOccurred())
-	Expect(crd.Spec.Names.Kind).To(Equal("KafkaTopic"))
+		err = k8sClient.Get(context.TODO(), types.NamespacedName{Name: "kafkatopics.kafka.banzaicloud.io"}, crd)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(crd.Spec.Names.Kind).To(Equal("KafkaTopic"))
 
-	err = k8sClient.Get(context.TODO(), types.NamespacedName{Name: "kafkausers.kafka.banzaicloud.io"}, crd)
-	Expect(err).NotTo(HaveOccurred())
-	Expect(crd.Spec.Names.Kind).To(Equal("KafkaUser"))
+		err = k8sClient.Get(context.TODO(), types.NamespacedName{Name: "kafkausers.kafka.banzaicloud.io"}, crd)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(crd.Spec.Names.Kind).To(Equal("KafkaUser"))
 
-	close(done)
-}, 60)
+		close(done)
+	}()
+	Eventually(done, 240).Should(BeClosed())
```

Comment:
What's the benefit of running the body of `BeforeSuite` in a goroutine and pass in a `done` channel to verify that has finished in the given time window instead of just simply relying on the `timeout` implemented by `BeforeSuite`which would not require the creation and use of the `done` channel ?

Reply:
Following to upgrade to of ginko and gomega, there was a WARNING reported pointing to: 
https://github.com/onsi/ginkgo/blob/v2/docs/MIGRATING_TO_V2.md#removed-async-testing

So they plan to remove the support for `BeforeSuite(func(done Done) {` and instead ignore the timeout if done that in the upcoming 2.0
Thought we should be proactive and refactor that

Reply:
I wonder if this can be simplified to:
```
BeforeSuite(func() {
...
....
}, 240)
```
in order to get rid of introducing a go routine and a channel.

Reply:
good call - updated